src/main/java/com/conveyal/r5/profile/FastRaptorWorker.java

```diff
@@ -0,0 +1,559 @@
+        runningScheduledPatterns = IntStream.of(originalPatternIndexForScheduledIndex)
+                .mapToObj(transit.tripPatterns::get).toArray(TripPattern[]::new);
+
+        LOG.info("Prefiltering patterns based on date active reduced {} patterns to {} frequency and {} scheduled patterns",
+                transit.tripPatterns.size(), frequencyPatterns.size(), scheduledPatterns.size());
+    }
+
+    /** Set the departure time in the scheduled search to the given departure time */
+    private void updateDepartureTime (int departureTime) {
+        for (RaptorState state : this.scheduleState) {
+            state.setDepartureTime(departureTime);
+
+            // clear all touched stops to avoid constant reëxploration
+            state.bestStopsTouched.clear();
+            state.nonTransferStopsTouched.clear();
+            // TODO prune trips that are now longer than max lengths to avoid biasing averages
+        }
+
+
+        // add initial stops
+        RaptorState initialState = scheduleState[0];
+        accessStops.forEachEntry((stop, accessTime) -> {
+            initialState.setTimeAtStop(stop, accessTime + departureTime, -1, -1, 0, 0, true);
+            return true; // continue iteration
+        });
+    }
+
+    /** Perform one minute of a RAPTOR search */
+    private int[][] runRaptorForMinute(int iterationsPerMinute) {
+        // Run the scheduled search
+        // round 0 is the street search
+        if (transit.hasSchedules) {
+            long startTime = System.nanoTime();
+            for (int round = 1; round <= request.maxRides; round++) {
+                // NB since we have transfer limiting not bothering to cut off search when there are no more transfers
+                // as that will be rare and complicates the code grabbing the results
+
+                // prevent finding crazy multi-transfer ways to get somewhere when there is a quicker way with fewer
+                // transfers
+                scheduleState[round].min(scheduleState[round - 1]);
+
+                long scheduledStartTime = System.nanoTime();
+                doScheduledSearchForRound(scheduleState[round - 1], scheduleState[round]);
+                timeInScheduledSearchTransit += System.nanoTime() - scheduledStartTime;
+
+                // perform a frequency search using worst-case boarding time to provide a tighter upper bound
+                long frequencyStartTime = System.nanoTime();
+                doFrequencySearchForRound(scheduleState[round - 1], scheduleState[round], true);
+                timeInScheduledSearchFrequencyBounds += System.nanoTime() - frequencyStartTime;
+
+                long transferStartTime = System.nanoTime();
+                doTransfers(scheduleState[round]);
+                timeInScheduledSearchTransfers += System.nanoTime() - transferStartTime;
+            }
+            timeInScheduledSearch += System.nanoTime() - startTime;
+        }
+
+        if (transit.hasFrequencies) {
```

Comment:
Maybe tack a comment in here to remind the reader that the frequency search is building on the results of the scheduled search at this particular minute.